File: Server/ChatServer/src/main/java/com/szlazakm/chatserver/dtos/response/OutputEncryptedMessageDTO.java
package com.szlazakm.chatserver.dtos.response;

import com.szlazakm.chatserver.dtos.EncryptedMessageDTO;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Value;
import lombok.experimental.SuperBuilder;

import java.util.UUID;

@Value
@Builder
public class OutputEncryptedMessageDTO {
    boolean initial;
    UUID id;
    String from;
    String to;
    String cipher;
    String aliceIdentityPublicKey;
    String aliceEphemeralPublicKey;
    Integer bobOpkId;
    Integer bobSpkId;
    String ephemeralRatchetKey;
    int messageIndex;
    int lastMessageBatchSize;
    String date;
}

File: client/app/src/main/java/com/szlazakm/safechat/webclient/dtos/OutputEncryptedMessageDTO.kt
package com.szlazakm.safechat.webclient.dtos

import com.fasterxml.jackson.annotation.JsonProperty
import java.util.UUID

data class OutputEncryptedMessageDTO(
    @JsonProperty("id")
    val id: UUID,
    @JsonProperty("initial")
    val initial: Boolean,
    @JsonProperty("from")
    val from: String,
    @JsonProperty("to")
    val to: String,
    @JsonProperty("cipher")
    val cipher: String,
    @JsonProperty("aliceIdentityPublicKey")
    val aliceIdentityPublicKey: String?,
    @JsonProperty("aliceEphemeralPublicKey")
    val aliceEphemeralPublicKey: String?,
    @JsonProperty("bobOpkId")
    val bobOpkId: Int?,
    @JsonProperty("bobSpkId")
    val bobSpkId: Int?,
    @JsonProperty("date")
    val date: String
)